File: src3/LoginManager.java
// In UserInterface.java or a new class for authentication
public class LoginManager {
    // Student login
    public static Student loginStudent(String studentID, String password) {
        for (Student student : ObjectCreator.students) {
            if (student.getStudentID().getId().equals(studentID) && student.getPassword().getPassword().equals(password)) {
                return student;
            }
        }
        return null; // Invalid credentials
    }

    // Advisor login
    public static Advisor loginAdvisor(String lecturerID, String password) {
        for (Advisor advisor : ObjectCreator.advisors) {
            if (advisor.getLecturerID().getId().equals(lecturerID) && advisor.getPassword().getPassword().equals(password)) {
                return advisor;
            }
        }
        return null; // Invalid credentials
    }
}

File: Iteration3/Python/LoginManager.py
class LoginManager:
    @staticmethod
    def login_student(student_id, password):
        for student in ObjectCreator.students:
            if student.get_student_id().get_id() == student_id and student.get_password().get_password() == password:
                return student
        return None  # Invalid credentials

    @staticmethod
    def login_advisor(lecturer_id, password):
        for advisor in ObjectCreator.advisors:
            if advisor.get_lecturer_id().get_id() == lecturer_id and advisor.get_password().get_password() == password:
                return advisor
        return None  # Invalid credentials
